lib/public/IAppConfig.php

```diff
@@ -29,44 +32,390 @@
+	 *
+	 * @return float stored config value or $default if not set in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy grouping
+	 * @see self::getValueString()
+	 * @see self::getValueInt()
+	 * @see self::getValueBool()
+	 * @see self::getValueArray()
+	 */
+	public function getValueFloat(string $app, string $key, float $default = 0, bool $lazy = false): float;
+
+	/**
+	 * Get config value assigned to a config key.
+	 * If config key is not found in database, default value is returned.
+	 * If config key belongs to a lazy group, the name of the lazy group needs to be specified.
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param bool $default default value (optional)
+	 * @param string $lazy name of the lazy group (optional)
+	 *
+	 * @return bool stored config value or $default if not set in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy grouping
+	 * @see self::getValueString()
+	 * @see self::getValueInt()
+	 * @see self::getValueFloat()
+	 * @see self::getValueArray()
+	 */
+	public function getValueBool(string $app, string $key, bool $default = false, bool $lazy = false): bool;
+
+	/**
+	 * Get config value assigned to a config key.
+	 * If config key is not found in database, default value is returned.
+	 * If config key belongs to a lazy group, the name of the lazy group needs to be specified.
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param array $default default value
+	 * @param bool $lazy search within lazy loaded config
+	 *
+	 * @return array stored config value or $default if not set in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy grouping
+	 * @see self::getValueString()
+	 * @see self::getValueInt()
+	 * @see self::getValueFloat()
+	 * @see self::getValueBool()
+	 */
+	public function getValueArray(string $app, string $key, array $default = [], bool $lazy = false): array;
+
+	/**
+	 * Store a config key and its value in database
+	 * If config key is already known with the exact same config value, the database is not updated.
+	 * If config key is not supposed to be read during the boot of the cloud, it is advised to set it as lazy loaded.
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param string $value config value
+	 * @param bool|null $sensitive value should be hidden when needed. if NULL sensitive flag is not changed in database
+	 * @param string $lazy name of the lazy group
+	 *
+	 * @return bool TRUE if value was different, therefor updated in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy grouping
+	 * @see self::setValueInt()
+	 * @see self::setValueFloat()
+	 * @see self::setValueBool()
+	 * @see self::setValueArray()
+	 */
+	public function setValueString(string $app, string $key, string $value, bool $lazy = false, ?bool $sensitive = null): bool;
```

Comment:
New method, let's force developers to do it properly.
Sensitive can not really have 3 states. Either it is sensitive or not.

```suggestion
	public function setValueString(string $app, string $key, string $value, bool $lazy = false, bool $sensitive = false): bool;
```

I would even argue that security should not be optional and neither should be the sensitive flag?

```suggestion
	public function setValueString(string $app, string $key, string $value, bool $sensitive, bool $lazy = false): bool;
```